tasks/README.md

```diff
@@ -104,8 +109,28 @@ If a parameter type accepts `null` the task runner will accept either a `null` v
 
 ### Metaparameters
 
-In addition to the tasks parameters, the task runner may inject metaparameters prefixed by '_'. These include `_noop` and `_task`.
+In addition to the tasks parameters, the task runner may inject metaparameters prefixed by `_`.
+
+**_noop**: Used to implement logic in tasks to support cases where task should not perform certain actions.
+
+**_task**: Allow multiple task implementations to access the same executable file. The `_task` metaparameter provides the executable the task name to allow task specific logic to be implemented. 
+
+**_installdir**: Tasks with `files` specified in the metadata will be passed the `_installdir` metaparameter to provide the file path to the expected resources. 
 
+#### Metaparameter Example
+
+When the task runner runs a task with `files` metadata it copies the specified files into a temporary directory on the target. The directory structure of the specified file resources will be preserved such that paths specified with the `files` metadata option will be available to tasks prefixed with `_installdir`. 
+
+###### Metadata
+```json
+{ "files": ["mymodule/files/script1.py"] }
+```
+###### Task
+```python
+require json
```

Comment:
This isn't actual Python, it's Ruby written in a Python style. Should be something like
```
import json, sys, os
params = json.load(sys.stdin)
os.execl([params['_installdir']+'/mymodule/files/script1.py', '--flag', params['flag1'], '-f', params['flag2'], params['arg1']])
```